btf/core.go

```diff
@@ -169,17 +169,26 @@ func (k coreKind) String() string {
 //
 // Fixups are returned in the order of relos, e.g. fixup[i] is the solution
 // for relos[i].
-func CORERelocate(relos []*CORERelocation, target *Spec, bo binary.ByteOrder, resolveLocalTypeID func(Type) (TypeID, error)) ([]COREFixup, error) {
-	if target == nil {
-		var err error
-		target, _, err = kernelSpec()
+func CORERelocate(relos []*CORERelocation, targets []*Spec, kmodName string, bo binary.ByteOrder, resolveLocalTypeID func(Type) (TypeID, error)) ([]COREFixup, error) {
+	if len(targets) == 0 {
```

Comment:
With https://github.com/cilium/ebpf/commit/4ad6b8c1eed691c5628580c922c3534509d67a70 in place this code can go into `applyRelocations` in package ebpf, since calling LoadKernelSpec is cheap again. That means we don't need to deal with kmodName here.

Reply:
Isn't that a behavior change? Since `CORERelocate` is public, calling with `targets = nil` means it will auto-load kernel BTF if available. Moving this code into `applyRelocations` would change that.

Reply:
Yes, but it's one I'm comfortable with. Either way we'll break the API. We've gone back and forth on this a couple of times already xD